packages/react/src/alert-dialog/trigger/AlertDialogTrigger.tsx

```diff
@@ -29,24 +28,17 @@ export const AlertDialogTrigger = React.forwardRef(function AlertDialogTrigger(
     [disabled, open],
   );
 
-  const mergedRef = useForkRef(forwardedRef, setTriggerElement);
-
-  const { getButtonProps } = useButton({
+  const { getButtonProps, buttonRef } = useButton({
     disabled,
-    buttonRef: mergedRef,
+    buttonRef: forwardedRef,
   });
 
-  const { renderElement } = useComponentRenderer({
-    render: render ?? 'button',
-    className,
+  return useRenderElement('button', componentProps, {
     state,
-    propGetter: (externalProps) => getButtonProps(getTriggerProps(externalProps)),
-    extraProps: other,
+    ref: [forwardedRef, buttonRef, setTriggerElement],
     customStyleHookMapping: triggerOpenStateMapping,
-    ref: mergedRef,
+    props: [elementProps, getTriggerProps, getButtonProps],
```

Comment:
The `getTriggerProps`/`getFloatingProps` getters should be already-called in the array, before `elementProps`